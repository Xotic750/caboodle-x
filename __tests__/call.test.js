import {call} from '../dist/caboodle-x';
import nonSymbols from './helpers/nonSymbols';

describe('call', () => {
  const receiver = {};

  it('throws for non-functions', () => {
    const throwsIfNotCallable = function (notFunc) {
      expect(() => call(notFunc, receiver)).toThrowErrorMatchingSnapshot();
    };

    nonSymbols.forEach(throwsIfNotCallable);
  });

  it('has correct context and arguments', () => {
    call(
      function (a, b) {
        expect(this).toBe(receiver, 'context matches expected');
        expect([a, b]).toEqual([1, 2], 'named args are correct');
        expect(arguments).toHaveLength(3, 'extra argument was passed');
        expect(arguments[2]).toEqual(3, 'extra argument was correct');
      },
      receiver,
      1,
      2,
      3,
    );
  });
});
