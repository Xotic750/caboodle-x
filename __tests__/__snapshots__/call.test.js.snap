// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`call throws for non-functions 1`] = `"Not a function"`;

exports[`call throws for non-functions 2`] = `"Not a function"`;

exports[`call throws for non-functions 3`] = `"Not a function"`;

exports[`call throws for non-functions 4`] = `"Not a function"`;

exports[`call throws for non-functions 5`] = `"Not a function"`;

exports[`call throws for non-functions 6`] = `"Not a function"`;

exports[`call throws for non-functions 7`] = `"Not a function"`;

exports[`call throws for non-functions 8`] = `"Not a function"`;

exports[`call throws for non-functions 9`] = `"Not a function"`;

exports[`call throws for non-functions 10`] = `"Not a function"`;

exports[`call throws for non-functions 11`] = `"Not a function"`;

exports[`call throws for non-functions 12`] = `"Not a function"`;

exports[`call throws for non-functions 13`] = `"Not a function"`;

exports[`call throws for non-functions 14`] = `"Not a function"`;

exports[`call throws for non-functions 15`] = `"Not a function"`;

exports[`call throws for non-functions 16`] = `"Not a function"`;

exports[`call throws for non-functions 17`] = `"Not a function"`;

exports[`call throws for non-functions 18`] = `"Not a function"`;

exports[`call throws for non-functions 19`] = `"Not a function"`;

exports[`call throws for non-functions 20`] = `"Not a function"`;

exports[`call throws for non-functions 21`] = `"Not a function"`;

exports[`call throws for non-functions 22`] = `"Not a function"`;

exports[`call throws for non-functions 23`] = `"Not a function"`;

exports[`call throws for non-functions 24`] = `"Not a function"`;

exports[`call throws for non-functions 25`] = `"Not a function"`;

exports[`call throws for non-functions 26`] = `"Not a function"`;

exports[`call throws for non-functions 27`] = `"Not a function"`;

exports[`call throws for non-functions 28`] = `"Not a function"`;

exports[`call throws for non-functions 29`] = `"Not a function"`;

exports[`call throws for non-functions 30`] = `"Not a function"`;

exports[`call throws for non-functions 31`] = `"Not a function"`;

exports[`call throws for non-functions 32`] = `"Not a function"`;

exports[`call throws for non-functions 33`] = `"Not a function"`;

exports[`call throws for non-functions 34`] = `"Not a function"`;

exports[`call throws for non-functions 35`] = `"Not a function"`;

exports[`call throws for non-functions 36`] = `"Not a function"`;

exports[`call throws for non-functions 37`] = `"Not a function"`;

exports[`call throws for non-functions 38`] = `"Not a function"`;

exports[`call throws for non-functions 39`] = `"Not a function"`;

exports[`call throws for non-functions 40`] = `"Not a function"`;

exports[`call throws for non-functions 41`] = `"Not a function"`;

exports[`call throws for non-functions 42`] = `"Not a function"`;

exports[`call throws for non-functions 43`] = `"Not a function"`;

exports[`call throws for non-functions 44`] = `"Not a function"`;

exports[`call throws for non-functions 45`] = `"Not a function"`;

exports[`call throws for non-functions 46`] = `"Not a function"`;

exports[`call throws for non-functions 47`] = `"Not a function"`;

exports[`call throws for non-functions 48`] = `"Not a function"`;

exports[`call throws for non-functions 49`] = `"Not a function"`;

exports[`call throws for non-functions 50`] = `"Not a function"`;

exports[`call throws for non-functions 51`] = `"Not a function"`;

exports[`call throws for non-functions 52`] = `"Not a function"`;

exports[`call throws for non-functions 53`] = `"Not a function"`;

exports[`call throws for non-functions 54`] = `"Not a function"`;
